ambry-store/src/main/java/com.github.ambry.store/DefaultCompactionPolicy.java

```diff
@@ -39,7 +39,7 @@
   DefaultCompactionPolicy(StoreConfig storeConfig, Time time) {
     this.storeConfig = storeConfig;
     this.time = time;
-    this.messageRetentionTimeInMs = storeConfig.storeDeletedMessageRetentionDays * Time.SecsPerDay * Time.MsPerSec;
+    this.messageRetentionTimeInMs = (long) storeConfig.storeDeletedMessageRetentionDays * Time.SecsPerDay * Time.MsPerSec;
```

Comment:
Same here. This is being fixed as part of [this](https://github.com/linkedin/ambry/pull/617/files#diff-d4308449279cec206916ab71281fe031L42).